### PR TITLE
Convert docs to OCaml format only.

### PR DESCRIPTION
**Pull request by jdeisenberg**
Following the pattern established for `Float` and `Int`, I have removed the Reason documentation and converted from markdown to `odoc` format.

This is done in anticipation of a time when `odoc` or a similar program will be able to extract sample code and convert it to Reason during the build process.

### CODE REVIEW

**Comment by Dean177** [comment]
Looks like it’s on the horizon for odoc:

https://github.com/ocaml/odoc/issues/129

And it’s getting some support to gain this feature

https://www.reason-association.org/projects/

**Comment by jdeisenberg** [comment]
The preceding “consolidate“ commit groups together individual `{[ ... ]}` examples for a function into a single `{[...]}` block.

It also puts the documentation with the BuckleScript snake_case version of functions, and the ReasonML camelCase version becomes `Same as {!Module.see_other}.`

**Comment by pbiggar** [comment]
I know I have this in email somewhere, but for the benefit of others reading this, can you comment about why you think this is the right plan and what the future of this is?

Thanks!

**Comment by jdeisenberg** [comment]
In a conversation with some people involved with odoc, we came to the conclusion that the easiest things to do are to recognize code within examples and extract them to create verifiable tests as part of the document build process. Another step in the process would be to convert the example docs to ReasonML

File: bs/src/tablecloth.mli [hunk 1]
```diff
@@ -7,71 +7,49 @@ Function names that are in snake_case have their documentation written in OCaml
...
 (**
-    The `>>` operator returns a function that is the equivalent of the composition of its function arguments.
-    The main use of `>>` is to avoid writing parentheses.
-
-  `(f >> g) x` (`(f >> g)(x)` in ReasonML) is the equivalent of `f (g x)` (`f(g(x))` in ReasonML)
+    The [>>] operator returns a function that is the equivalent of the composition of its function arguments.
+    The main use of [>>] is to avoid writing parentheses.
 
-  ### Example
+  [(f >> g) x] is the equivalent of [f (g x)]
```
Comment (wolfgangGoedel) [inline bs/src/tablecloth.mli]: This should be `[(f >> g) x] is the equivalent of [g (f x)]`

File: bs/src/tablecloth.mli [hunk 2]
```diff
@@ -7,71 +7,49 @@ Function names that are in snake_case have their documentation written in OCaml
...
 
+val ( << ) : ('b -> 'c) -> ('a -> 'b) -> 'a -> 'c
 (**
-  The `<<` operator returns a function that is the equivalent of the reverse composition of its function arguments.
-
-  `(f << g) x` (`(f << g)(x)` in ReasonML) is the equivalent of `g (f x)` (`g(f(x))` in ReasonML)
+  The [<<] operator returns a function that is the equivalent of the reverse composition of its function arguments.
 
-  ### Example
+  [(f << g) x] is the equivalent of [g (f x)]
```
Comment (wolfgangGoedel) [inline bs/src/tablecloth.mli]: ...and this the other way: `[(f << g) x] is the equivalent of [f (g x)]`
The docs interchanged signatures for forward and backward composition

**Comment by pbiggar** [comment]
Apologies for the radio silence, i missed this. @jdeisenberg - can you rebase and handle @wolfgangGoedel's comments? Thanks!

**Comment by jdeisenberg** [comment]
I'm not sure how to do a rebase. I have never done one successfully, and always end up removing all the files, cloning again, and starting over.

**Comment by pbiggar** [comment]
I rebased this and fixed the merge conflicts. Can you check over to make sure it's right (the conflicts were in List.concat, which had moved in the file).

Merging this now to avoid future conflicts.

**Pull request merged**